### PR TITLE
Added some reasonable defaults to LIRC's module

### PR DESCRIPTION
**Pull request by alex-ameen**
###### Motivation for this change

After spending a few hours trying to install LIRC I realized that many of my issues were from a lack of default settings that could get the program up and running. This will allow many users to let LIRC run out of the box.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/hardware/lirc.nix [hunk 1]
```diff
@@ -19,12 +19,29 @@ in {
...
+          output          = /run/lirc/lircd
+          pidfile         = /run/lirc/lircd.pid
+          plugindir       = ${pkgs.lirc}/lib/lirc/plugins
+          permission      = 666
+          allow-simulate  = No
+          repeat-max      = 600
+
+          [lircmd]
+          uinput          = False
+          nodaemon        = False
```
Comment (Mic92) [inline nixos/modules/services/hardware/lirc.nix]: To bad this was not added in the first place. I would prefer having sane default options. that can be extended by `extraOptions`.

Comment (Infinisil) [reply nixos/modules/services/hardware/lirc.nix]: I think for now this is alright, can always add that later

**Comment by Infinisil** [comment]
@BadDecisionsAlex We should strive to keep the default config as small as possible. If @ck3d's config works then that is preferred.

File: nixos/modules/services/hardware/lirc.nix (hunk 1 above)
Comment (Mic92) [reply nixos/modules/services/hardware/lirc.nix]: agreed.

**Comment by BadDecisionsAlex** [comment]
> My configuration file looks as follows:
> 
> ```
> [lircd]
> driver = ftdi
> device = product=0x6015
> ```
> I set none of the default values in my configuration file without any problems. Could you please specify which problems you have?

Sorry for the late reply, I've been swamped.
Without the default settings that I added LIRC was unable to access any of the drivers which it is normally packaged with.

**Comment by ck3d** [comment]
No problem.
To check that lircd find its drivers, call following command:
```
lircd -H help
```
In my case, everything is found. The default, ftdi and 30 more drivers.

I think the main reason is the missing device setting. Please give following config a try:
```
[lircd]
device = /dev/lirc0
```
or set it as extraArguments:
extraArguments = [ "--device=/dev/lirc0" ];

I had a look into lirc sources and it showed that the default device of the lircd default driver is /dev/lirc/0 (defined as LIRC_DRIVER_DEVICE). If /dev/lirc0 is a better default, we should think about patching lirc.

As alternative we could provide two NixOS options for driver and device.

**Comment by BadDecisionsAlex** [comment]
Sorry I did not mean to revive this. This has been sitting without interest for a while. I'm closing it.

**Pull request closed without merging**